reverse_engineering/api.js

```diff
@@ -66,38 +62,41 @@ module.exports = {
 	},
 
 	getDbCollectionsNames: async function (connectionInfo, logger, cb, app) {
-		logInfo('Retrieving labels information', connectionInfo, logger);
+		const step = 'Retrieving labels information';
+
+		logInfo(step, connectionInfo, logger);
 		try {
 			const sshService = app.require('@hackolade/ssh-service');
-			initDependencies(app);
 			neo4j.setTimeOut(connectionInfo);
 
 			await neo4j.connect(connectionInfo, checkConnection(logger), sshService);
 			logger.log('info', 'Successfully connected to the database instance', 'Connection');
 
 			const isMultiDb = await neo4j.supportsMultiDb();
+			logger.log('info', `Multi database support: ${isMultiDb}`, step);
 
 			const databaseNames = await neo4j.getDatabaseName('graph.db', isMultiDb);
-			logger.log('info', 'Name of database successfully retrieved', 'Retrieving labels information');
-			const result = await Promise.all(
-				databaseNames.map(async name => {
-					return {
-						dbName: name,
-						dbCollections: await neo4j.getLabels(name, isMultiDb),
-					};
-				}),
-			);
-			logger.log('info', 'Labels successfully retrieved', 'Retrieving labels information');
-			logger.log('info', 'Information about labels successfully retrieved', 'Retrieving labels information');
-			cb(null, result);
+			logger.log('info', 'Database names are successfully retrieved', step);
+
+			const results = [];
+			for (let databaseName of databaseNames) {
+				logger.log('info', `Fetching the labels from ${databaseName}`, step);
+				results.push({
+					dbName: databaseName,
+					dbCollections: await neo4j.getLabels({ databaseName, isMultiDb, logger }),
```

Comment:
Maybe it is worth catching possible errors here and returning the labels list only for successful DBs?